src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java

```diff
@@ -175,12 +177,14 @@ private void checkTenantNotExistsAndConsortiumExistsOrThrow(UUID consortiumId, S
       throw new ResourceAlreadyExistException("id", tenantId);
     }
   }
-
-  private void checkTenantAndConsortiumExistsOrThrow(UUID consortiumId, String tenantId) {
+  @Override
+  public void checkTenantsAndConsortiumExistsOrThrow(UUID consortiumId, List<String> tenantIds) {
     consortiumService.checkConsortiumExistsOrThrow(consortiumId);
-    if (!tenantRepository.existsById(tenantId)) {
-      throw new ResourceNotFoundException("id", tenantId);
-    }
+    tenantIds.forEach(tenantId -> {
```

Comment:
on prod we will have 61 tenants, so we can optimize this part to not do 61 separate requests to DB.
You can use findAllById(ids) method.
If size of this list is different that size of tenantIds -> we can throw an exception